.github/workflows/ci-more.yml

```
@@ -0,0 +1,52 @@
+on:
+  schedule:
+    # Run this script every saturday, so you have the whole weekend to fix problems :evil-grin:
+    - cron: '0 0 * * 6'
+jobs:
+  merge_into_ci_more:
+    # Only enable this job on dividelabs' repo to avoid running this job on forks
+    if: ${{ github.repository_owner == 'dividelabs' }}
+    runs-on: 'ubuntu-latest'
+    steps:
+      - name: 'Checkout repo'
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: 'Configure git'
+        run: |
+          git config --global user.name "ci-more cron job"
+          git config --global user.email "ci-more-cron-job@example.com"
+      - name: 'Checkout ci-more'
+        run: |
+          git checkout ci-more
+      - name: 'Backup .travis.yml and appveyor.yml'
+        run: |
+          cp .travis.yml .travis.yml.backup
+          cp appveyor.yml appveyor.yml.backup
+      - name: 'Checkout .travis.yml and appveyor.yml from master'
+        run: |
+          git checkout origin/master -- .travis.yml appveyor.yml
+          git add .travis.yml appveyor.yml
+          git commit -m "Temporarily add .travis.yml and appveyor.yml from master"
+      - name: 'Merge master into ci-more'
+        run: |
+          git merge origin/master -m "Merge master into ci-more"
+      - name: 'Restore .travis.yml and appveyor.yml'
+        run: |
+          mv .travis.yml.backup .travis.yml
+          mv appveyor.yml.backup appveyor.yml
+          git add -- .travis.yml appveyor.yml
+          git commit -m "Restore .travis.yml and appveyor.yml"
+      - name: 'Check whether the working directory is clean'
+        run: |
+          git diff --exit-code
+      - name: 'Check whether only .travis.yml and appveyor.yml differ with master'
+        run: |
+          if test $(git diff origin/master --name-only | wc -l) != 2; then
+            echo "ci-more should diverge from origin/master with exactly 2 files (.travis.yml and appveyor.yml)";
+            git diff origin/master --exit-code;  # origin/master
+            exit 1;
+          fi
```

When you end up removing `.travis.yml` and `appveyor.yml` from master, then `ci-more.yml` can be reduced.

```suggestion
      - name: 'Merge master into ci-more'
        run: |
          git merge origin/master -m "Merge master into ci-more"
      - name: 'Check whether the working directory is clean'
        run: |
          git diff --exit-code
      - name: 'Check whether only .travis.yml and appveyor.yml differ with master'
        run: |
          if test "$(git diff origin/master --name-only | xargs)" != ".travis.yml appveyor.yml"; then
              git diff origin/master --exit-code;
          fi
```

---

> Please check all options of the function, because once added (and released),
it's very hard (not to say impossible) to remove them.

It's hard for me to tell if this is implemented correctly, so I am going (mostly) for the documentation. I really don't know how CMake operates in this context.

Note, I might deprecate --schema and rename to --binary-schema or --reflection in a future flatcc version, but that doesn't change much here.

Looking at the comments in FlatccGenerateSources.cmake:

I wrote earlier that I'm not sure that setting READER as the default argument. But it also seems that ALL, RECURSIVE is being set, which I think is more useful:

> set(output_options BINARY_SCHEMA COMMON COMMON_READER COMMON_BUILDER BUILDER READER VERIFIER JSON_PARSER JSON_PRINTER JSON)
    set(NO_VAL_ARGS ALL RECURSIVE ${output_options})
    set(SINGLE_VAL_ARGS NAME OUTPUT_DIR OUTFILE PREFIX)
    set(MULTI_VAL_ARGS SCHEMA_FILES COMPILE_FLAGS PATHS)
    
   But later the following section says differently:
   
   > if(default_option_reader)
        set(FLATCC_READER ON)
    endif()
    
    
   There is an old error message here, should be "No flatbuffer schema files provided":
   
   > if (NOT FLATCC_SCHEMA_FILES)
        message(FATAL_ERROR "No flatbuffer definition files provided")
    endif()


I don't understand this line:

>  set(MULTI_VAL_ARGS SCHEMA_FILES COMPILE_FLAGS PATHS)

Later you append -I prefix to FLATCC_PATHS - I'm note sure what a keyword args on other CMake context.

Regarding the COMPILE_FLAGS option, this is (probably) correct,  but I wonder if EXTRA_ARGS, EXTRA_FLAGS, or something else would be more meaningful.

---

> Note, I might deprecate --schema and rename to --binary-schema or --reflection in a future flatcc version, but that doesn't change much here.

The function now accepts `BINARY_SCHEMA` only.
This function/file is meant to be distributed along flatcc, so it can be kept in sync.
When some options changes in the future, there are ways to ease the transition.

> Looking at the comments in FlatccGenerateSources.cmake:
> 
> I wrote earlier that I'm not sure that setting READER as the default argument. But it also seems that ALL, RECURSIVE is being set, which I think is more useful:

Is `ALL` the new agreed-upon default for cmake then?
You can also choose for a non-standard option e.g. `COMMON` + `BUILDER` + `VERIFIER` + `READER` without `RECURSIVE`.
But I think this will be harder to document, for no gain.

> > set(output_options BINARY_SCHEMA COMMON COMMON_READER COMMON_BUILDER BUILDER READER VERIFIER JSON_PARSER JSON_PRINTER JSON)
> > set(NO_VAL_ARGS ALL RECURSIVE ${output_options})
> > set(SINGLE_VAL_ARGS NAME OUTPUT_DIR OUTFILE PREFIX)
> > set(MULTI_VAL_ARGS SCHEMA_FILES COMPILE_FLAGS PATHS)
> 
> But later the following section says differently:
> 
> > if(default_option_reader)
> > set(FLATCC_READER ON)
> > endif()

What this does, is iterate over all  output_options (`ALL BINARY_SCHEMA COMMON COMMON_READER COMMON_BUILDER BUILDER READER`) and if none of these are set, then assume the caller has passed `READER` (=default output option when no output has been selected).

> There is an old error message here, should be "No flatbuffer schema files provided":
> 
> > if (NOT FLATCC_SCHEMA_FILES)
> > message(FATAL_ERROR "No flatbuffer definition files provided")
> > endif()

Fixed (also removed all reference of definition file to schema file).


> I don't understand this line:
> 
> > set(MULTI_VAL_ARGS SCHEMA_FILES COMPILE_FLAGS PATHS)

This variable is later used in `cmake_parse_arguments`.
It is a list of arguments that except multiple arguments.
`SCHEMA_FILES` can be one file or more.

> Later you append -I prefix to FLATCC_PATHS - I'm note sure what a keyword args on other CMake context.
> 
> Regarding the COMPILE_FLAGS option, this is (probably) correct, but I wonder if EXTRA_ARGS, EXTRA_FLAGS, or something else would be more meaningful.

I copied the terminology from [`FindBISON`](https://cmake.org/cmake/help/latest/module/FindBISON.html). But I guess `EXTRA_FLAGS` is ok too.

---

I created another PR to @madebr 's branch.

Things still to do/check:
- When running cmake with BUILD_SHARED_LIBS=On the installed flatcc exe can't find the flatcc compiler shared lib (on Linux) unless you set LD_LIBRARY_PATH. I think we should add RPATH.
- Cross-compiling and Conan compatibility  to be checked

If you can postpone the merge to Tuesday or so, I can do that.

---

@wdobbe

I'm in no hurry to merge.

@madebr

If the READER is default, what does the following mean?
> set(NO_VAL_ARGS ALL RECURSIVE ${output_options})

> Is ALL the new agreed-upon default for cmake then?

No I am raising the discussion. I would expect at least READER + COMMON_READER as default because without a COMMON_READER the result will not be usable unless the common file is created elsewhere.

But in general ALL + RECURSIVE will build anything a user would expect when not caring too much about arguments, which is why I think that will be a good choice. But I am open to suggestions.

It looks to me as if you expand all the -I  paths without the -I prefix and later try to parse it, but I am probable just not understanding the mechanism:

> set(MULTI_VAL_ARGS SCHEMA_FILES COMPILE_FLAGS PATHS)
> This variable is later used in cmake_parse_arguments.

---

> I created another PR to @madebr 's branch.

Thanks! And merged!

>     * When running cmake with BUILD_SHARED_LIBS=On the installed flatcc exe can't find the flatcc compiler shared lib (on Linux) unless you set LD_LIBRARY_PATH. I think we should add RPATH.

This is added, and tested.
See https://github.com/madebr/flatcc/actions/runs/432848623


>     * Cross-compiling and Conan compatibility  to be checked

I'll leave that up to you :D